### ISSUE TITLE
Unable to open Report using Point Type as a filter after upgrade 2.11->2.12

### ISSUE THREAD

**Issue by isleshocky77** [mautic/mautic#5683]
**Please DO NOT report security vulnerabilities here. Send them to security@mautic.com instead.**

What type of report is this:

| Q  | A
| ---| ---
| Bug report? | Yes

## Description:

After upgrading to 2.12 I'm no longer able to open my reports which have Point Type as a filter. Get the error 

```
PHP Notice:  Undefined index: lp.type in /srv/mautic/project/app/bundles/ReportBundle/Form/Type/DynamicFiltersType.php on line 32,
```

When trying to create a new Report it looks like that field is no longer in the drop down for filters.

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.12.0
| PHP version | 5.6.33

 
### Log errors: 

```
PHP Notice:  Undefined index: lp.type in /srv/mautic/project/app/bundles/ReportBundle/Form/Type/DynamicFiltersType.php on line 32,
```

**Comment by isleshocky77** [comment]
This is the commit in question which causes the issue

```
ee67e150217de5bce97f8e17889060e15d119580 is the first bad commit
commit ee67e150217de5bce97f8e17889060e15d119580
Author: Alan Hartless <alan@devkardia.com>
Date:   Wed Nov 22 11:54:13 2017 -0600

    Fixed missing company fields from lead reports and add IP lead xref left join method

:040000 040000 29fa93891451acd3f07b809500ec3ec6f1e6c9f9 5c6e45967fabaa05534d4efafeff34dcede31686 M	app
```

**Comment by isleshocky77** [comment]
It appears that @alanhartless forgot to merge the current `$filters` with the new columns in the `injectXX` method.

```diff
index 03c72e6..c2b8d02 100644
--- a/app/bundles/LeadBundle/EventListener/ReportSubscriber.php
+++ b/app/bundles/LeadBundle/EventListener/ReportSubscriber.php
@@ -678,7 +678,7 @@ class ReportSubscriber extends CommonSubscriber
         $data = [
             'display_name' => 'mautic.lead.report.points.table',
             'columns'      => array_merge($columns, $pointColumns, $event->getIpColumn()),
-            'filters'      => $filters,
+            'filters'      => array_merge($filters, $pointColumns, $event->getIpColumn()),
         ];
         $event->addTable(self::CONTEXT_LEAD_POINT_LOG, $data, self::GROUP_CONTACTS);
 
@@ -733,7 +733,7 @@ class ReportSubscriber extends CommonSubscriber
         $data = [
             'display_name' => 'mautic.lead.report.frequency.messages',
             'columns'      => array_merge($columns, $frequencyColumns),
-            'filters'      => $filters,
+            'filters'      => array_merge($filters, $frequencyColumns),
         ];
         $event->addTable(self::CONTEXT_CONTACT_FREQUENCYRULES, $data, self::GROUP_CONTACTS);
     }
```

**Comment by isleshocky77** [comment]
Looks like this issue is solved in staging on commit 583a0fc4f8731e2a7c71aa99c5eafcf375c3ee86 and #5601
